### PR TITLE
Applied necessary changes to docs and docker files for Jazzy

### LINKED ISSUE #266: Add Jazzy support
### Summary

- [x] Create Jazzy branch
- [ ] Create docker setup for Jazzy
- [ ] Migrate packages to use ROS Jazzy
- [ ] Deprecate andino_gz_classic package
- [ ] Test on real robot.

### PR DESCRIPTION
**Pull request by JesusSilvaUtrera**
# 🎉 New feature

Closes #266 

## Summary
This PR adds to the `jazzy` branch the necessary changes for the code to work on Jazzy, both with docker and directly Ubuntu 24.

## Test it
Follow the instructions on the README to test both locally or with the docker environment.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -15,32 +14,37 @@ RUN apt-get update && apt-get install --no-install-recommends -y $(cat requireme
...
-# Adds colcon autocomplete
-RUN echo "source /usr/share/colcon_argcomplete/hook/colcon-argcomplete.bash" >> /home/$USER/.bashrc
-# Adds gazebo sourcing.
-RUN echo "source /usr/share/gazebo/setup.bash" >> /home/$USER/.bashrc
+# Adds to bashrc the ros jazzy overlay sourcing.
+RUN echo "source /opt/ros/jazzy/setup.bash" >> /home/$USER/.bashrc
 
 # Updates
-RUN sudo apt upgrade -y && sudo apt update && rosdep update
+RUN sudo apt update && sudo apt upgrade -y && rosdep update
```
Comment (jballoffet) [inline docker/Dockerfile:42]: Shall this change be back-ported to Humble branch?

File: docker/run.sh [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 # BSD 3-Clause License
 #
-# Copyright (c) 2023, Ekumen Inc.
+# Copyright (c) 2024, Ekumen Inc.
```
Comment (jballoffet) [inline docker/run.sh:5]: Ditto.

File: docker/build.sh [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 # BSD 3-Clause License
 #
-# Copyright (c) 2023, Ekumen Inc.
+# Copyright (c) 2024, Ekumen Inc.
```
[the remark this answers is not in the document]
Comment (JesusSilvaUtrera) [reply docker/build.sh:5]: Oops, my bad

File: docker/Dockerfile (hunk 1 above)
Comment (JesusSilvaUtrera) [reply docker/Dockerfile:42]: I don't think so, it's just that for some reason if I don't do the uodate first on Ubuntu 24 some repositories are not found, in the hunvle branch it has been working fine until now.

**Review by jballoffet — APPROVED** [review]
LGTM, thanks @JesusSilvaUtrera!

File: docker/Dockerfile (hunk 1 above)
Comment (jballoffet) [reply docker/Dockerfile:42]: Ack.

**Pull request merged**